server/pulp/server/webservices/views/util.py

```diff
@@ -111,3 +111,25 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 json_body_allow_empty = functools.partial(json_body_required, allow_empty=True)
+
+
+def page_not_found(request, *args, **kwargs):
+    """
+    Returns a HttpResponse with an empty json payload and a 404 response code.
+
+    This function uses *args and **kwargs to be compatible with newer versions of Django which
+    have additional parameters.
+
+    :param request: The WSGI object containing the request
+    :type request: django.core.handlers.wsgi.WSGIRequest
+    :param args: Additional positional arguments
+    :type args: list
+    :param kwargs: Additional keyword arguments
+    :type kwargs: dict
+
+    :return: A response object that contains a 404 status code and an empoty json payload.
+    :rtype: django.http.HttpResponse
+    """
+    json_response = generate_json_response()
+    json_response.status_code = 404
```

Comment:
Good idea. I updated this line, the rest of this module, and the associated tests to use `httplib`.